ignite/internal/tools/gen-mig-docs/main.go

```diff
@@ -0,0 +1,630 @@
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"io/fs"
+	"log"
+	"os"
+	"path/filepath"
+	"sort"
+	"strings"
+
+	"github.com/Masterminds/semver/v3"
+	"github.com/go-git/go-git/v5"
+	"github.com/go-git/go-git/v5/plumbing"
+	"github.com/hexops/gotextdiff"
+	"github.com/hexops/gotextdiff/myers"
+	"github.com/hexops/gotextdiff/span"
+	"github.com/ignite/cli/ignite/pkg/cmdrunner/exec"
+	"github.com/ignite/cli/ignite/pkg/cmdrunner/step"
+	"github.com/pkg/errors"
+)
+
+const (
+	igniteCliRepository = "http://github.com/ignite/cli.git"
+	igniteBinaryPath    = "dist/ignite"
+	igniteRepoPath      = "src/github.com/ignite/cli"
+)
+
+var scaffoldCommands = map[string][]string{
+	"chain":  {"chain example --no-module"},
+	"module": {"chain example"},
+	"list": {
+		"chain example",
+		"list list1 f1:string f2:strings f3:bool f4:int f5:ints f6:uint f7:uints f8:coin f9:coins --module example --yes",
+	},
+	"map": {
+		"chain example",
+		"map map1 f1:string f2:strings f3:bool f4:int f5:ints f6:uint f7:uints f8:coin f9:coins --module example --yes",
+	},
+	"single": {
+		"chain example",
+		"single single1 f1:string f2:strings f3:bool f4:int f5:ints f6:uint f7:uints f8:coin f9:coins --module example --yes",
+	},
+	"type": {
+		"chain example",
+		"type type1 f1:string f2:strings f3:bool f4:int f5:ints f6:uint f7:uints f8:coin f9:coins --module example --yes",
+	},
+	"message": {
+		"chain example",
+		"message message1 f1:string f2:strings f3:bool f4:int f5:ints f6:uint f7:uints f8:coin f9:coins --module example --yes",
+	},
+	"query": {
+		"chain example",
+		"query query1 f1:string f2:strings f3:bool f4:int f5:ints f6:uint f7:uints --module example --yes",
+	},
+	"packet": {
+		"chain example --no-module",
+		"module example --ibc",
+		"packet packet1 f1:string f2:strings f3:bool f4:int f5:ints f6:uint f7:uints f8:coin f9:coins --module example --yes",
+	},
+}
+
+var diffExceptions = []string{
+	"*.md",
+	"go.sum",
+	"*_test.go",
+	"*.pb.go",
+	"*.pb.gw.go",
+	"openapi.yml", // FIXME: For some reason removing this file from exceptions causes too much memory usage and OS kills the process
```

Comment:
```suggestion
	"**/node_modules/", 
	"openapi.yml", // FIXME: For some reason removing this file from exceptions causes too much memory usage and OS kills the process
```

Reply:
Here we use [filepath.Match](https://pkg.go.dev/path/filepath#Match) function which i believe only works with simple filename patterns and doesn't work with directory patterns and ...

Reply:
Nvm, i think this can be solved with github.com/gobwas/glob